File: app/src/main/java/br/com/diogomoreno/events/data/datasource/remote/event/EventRemoteDataSource.java
package br.com.diogomoreno.events.data.datasource.remote.event;


import java.util.List;

import javax.inject.Inject;

import br.com.diogomoreno.events.domain.model.Event;
import retrofit2.Call;

public class EventRemoteDataSource {

    private final EventService eventService;

    @Inject
    public EventRemoteDataSource(EventService eventService) {
        this.eventService = eventService;
    }

    public Call<List<Event>> getEvents(){
        return eventService.getEvents();
    }

    public Call<Event> getEvent(int eventId){
        return eventService.getEvent(eventId);
    }
}

File: app/src/main/java/br/com/diogomoreno/events/data/datasource/remote/event/EventRemoteDataSource.kt
package br.com.diogomoreno.events.data.datasource.remote.event

import br.com.diogomoreno.events.domain.model.Event
import retrofit2.Call
import javax.inject.Inject

class EventRemoteDataSource @Inject constructor(private val eventService: EventService) {
    val events: Call<MutableList<Event?>?>
        get() = eventService.events

    fun getEvent(eventId: Int): Call<Event?>? {
        return eventService.getEvent(eventId)
    }

}